packages/transaction-controller/src/utils/swaps.ts

```diff
@@ -220,7 +210,9 @@ export async function updatePostTransactionBalance(
       transactionMeta.txParams.from,
     ]);
 
-    latestTransactionMeta = getTransaction(transactionId) as TransactionMeta;
+    latestTransactionMeta = {
+      ...(getTransaction(transactionId) ?? ({} as TransactionMeta)),
```

Comment:
spreading `undefined` is valid. is the nullish coalesce needed?

Reply:
Yeah, TypeScript gets confused for some reason if I just simplify this to `...getTransaction(transactionId)`. It says:

```
[tsserver 2322] [E] Type '{} | { approvalTxId?: string | undefined; actionId?: string | undefined; baseFeePerGas?: `0x${string}` | undefined; blockNumber?: string | undefined; blockTimestamp?: string | undefined; ... 57 more ...; status: TransactionStatus.approved | ... 6 more ... | TransactionStatus.unapproved; } | { ...; }' is not assignable to type 'TransactionMeta | undefined'.
  Type '{}' is not assignable to type 'TransactionMeta | undefined'.
```

Reply:
There are two ways we could spread `undefined` here:

1. We could type `{}` as `Record<string, never>`
(Can't use the type `{}` as it refers to `NonNullable<unknown>`)
```ts
  let latestTransactionMeta:
    | TransactionMeta
    | Record<string, never>
    | undefined;

  ...

    latestTransactionMeta = {
      ...getTransaction(transactionId),
    };
```

2. Ternary
```ts
    const transaction = getTransaction(transactionId);
    latestTransactionMeta =
      transaction !== undefined ? { ...transaction } : ({} as TransactionMeta);
```

Personally I think the `??` version is best for readability and accurate typing (`Record<string, never>` feels like unnecessary noise).

Reply:
This is fine as is then. Thank you for the explanation :)